controllers/pokemon.js

```diff
@@ -0,0 +1,167 @@
+const pokemonSchema = require('../models/pokemon');
+const axios = require('axios');
+const customError = require('../utils/customError')
+
+// here are all the implementations of the API
+
+// responsible for populating database
+module.exports.populateDatabase = async (req,res) => {
+
+    // we first wait to see how many pokemons we have in our DB
+    await pokemonSchema.countDocuments({}, async (err, count) => {
+        console.log(count);
+        if(count < 100){
+
+            // if we have less than 100 we start requesting data
+            const firstGeneration = await axios.get('https://pokeapi.co/api/v2/generation/1');
+            for(let i = 0; i < 100 - count; i++){
```

Comment:
Good that you found this third party lib.
It might be that it brings better performance but the idea was if you can make yourself these calls run in || with no other lib.
Hint you can have a look at: [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

Reply:
Thanks for the hint, I was able to replace the third party lib with Promise.all :D